src/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -45,7 +45,7 @@ public T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder)
         /// </returns>
         public IList<T> CollectionOfFake<T>(int numberOfFakes)
         {
-            var result = new List<T>();
+            var result = new SynchronizedCollection<T>();

```

Comment:
The list won't be used by FakeItEasy, only by the user, so I don't think it needs to be thread-safe. The user has no reason to assume that it is.